nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/AbstractKafkaProcessor.java

```diff
@@ -141,6 +166,10 @@
         SHARED_DESCRIPTORS.add(CLIENT_ID);
         SHARED_DESCRIPTORS.add(SECURITY_PROTOCOL);
         SHARED_DESCRIPTORS.add(KERBEROS_PRINCIPLE);
+        SHARED_DESCRIPTORS.add(SSL_KEY_PASSWORD);
```

Comment:
If it is possible without any hackery I am all for it. The main issue for these once is that they were always available as dynamic properties along with all other Kafka properties but since they are 'passwords' and we do not support "sensitive" attribute for dynamic properties they were added as first class properties.
While it may work with SSLContextService for this case, the correct way of fixing something like this would be to provide support for sensitivity for dynamic properties